openmc/tallies.py

```diff
@@ -1611,6 +1635,15 @@ def _align_tally_data(self, other):
         ----------
         other : Tally
             The tally to outer product with this tally
+        filter_product : str
```

Comment:
They should be `str` at this point, as `None` is only input to `hybrid_product` to indicate that the hybrid case discussed in the docstring where the choice of `entrywise` or `tensor` for nuclides and scores depends on whether the two tallies have the same nuclides and scores. The default case can be a bit confusing, so we could think of setting hard defaults if we think that would be easier to understand while maintaining the expected outcomes.